### ISSUE TITLE
blank Duo authentication frame for some users

### ISSUE THREAD

**Issue by awillman** [Unicon/cas-mfa#135]
I hope that some of you can help me and my colleagues diagnose and resolve a Duo-related problem.  We have successfully implemented Duo authentication in many places across our enterprise, including our CAS infrastructure.  However, some people encounter the following problem when attempting to authenticate to several of our CAS-enabled applications:

An affected user successfully authenticates to the primary CAS username/password prompts, but at the point where a frame would normally open with the Duo authentication prompts, nothing happens; the user is simply presented with a blank window, halting the authentication process.

The problem reliably happens to the affected users, regardless of the browser they use.  Most of the affected users are using Macs, but a few are using Windows.  We have seen the problem within the latest versions of IE, Safari, Firefox, and Chrome.  We haven't yet discovered any common feature shared by those that have been affected.

The problem has been somewhat resolved in only two cases so far:  One user supposedly resolved the problem themselves by changing the default language of their system from Simplified Chinese to English and rebooting, but the problem returned when they changed the default language back to Simplified Chinese.  The problem was supposedly resolved for another person by upgrading Java on their system and rebooting.  I don't understand yet how either of these would affect Duo, but am simply sharing the information reported by our support staff.

If any of you can shed any light on what may be happening, or offer any suggestions for diagnosing or resolving this problem, please do.  Thank you.

Alex

**Comment by mmoayyed** [comment]
Well the "Simple Chinese" problem is more likely. I know CAS3 had a few issues with other locales when it came to rendering pages. Can you duplicate the issue?

**Comment by awillman** [comment]
Thank you for your response.  I haven't been able to reproduce the problem myself yet, but it happens reliably to those who do encounter it.  However, I plan on working with a couple colleagues tomorrow to try reproducing it, now that we have a potential clue to its cause.

Our support staff recently started asking those who have the problem whether they were using the multi-language support on their computers.  10 have responded so far.  8 of those use the multi-language support, and the other 2 have an OS configured with a primary language other than English.

**Comment by awillman** [comment]
We have replicated the problem by either changing the default language of a system or browser to something other than English.  It appears that our CAS implementation is generally handling the other languages fine, but its Duo portion isn't.

I have implemented a temporary workaround of enforcing the use of "locale=en" when accessing /cas/login, but hope to find a longer-term solution that permits the use of other locales.
